docs: add Zulip guide (#10921)
stdlib-js/stdlib: docs/contributing/zulip_guide.md

<!--

@license Apache-2.0

Copyright (c) 2026 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

<!-- lint disable first-heading-level -->

> <!-- FIXME: the following two empty sections are merely to satisfy the linter for `expected-html-sections` which cannot, atm, be turned off -->

Reviewer: This FIXME shouldn't ship. If we restructure the file to match the other contributing docs (dropping the package-README sections), this comment and the empty `usage`/`examples` sections become unnecessary.

<section class="usage">

</section>

<!-- /.usage -->

<section class="examples">

</section>

> <!-- /.examples -->

Reviewer: This file uses the package README structure (`<section class="usage">`, `<section class="examples">`, `<section class="intro">`, `<section class="notes">`, `<!-- lint disable first-heading-level -->`) but no other file in `docs/contributing/` follows that pattern. Contributing docs use a simpler structure — see `branching.md`, `development.md`, `FAQ.md`, etc. as references.

We should restructure this to match the existing contributing doc conventions: license header, top-level heading, blockquote description, then content, ending with `<section class="links">` for link definitions.

# Zulip Guide

> Guide to the stdlib Zulip chat.

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

> This guide provides best practices and community norms for the [**stdlib Zulip chat**][stdlib-zulip]—the home of the stdlib community [announced][zulip-announcement] in December 2025. Anyone is welcome to join, or browse the web-public channels (which you can see without signing up). The community norms and guidelines here are to help ensure that everyone can get the most out of this platform and, hopefully, have a positive experience.

Reviewer 2: ```suggestion
This guide provides best practices and community norms for the [**stdlib Zulip chat**][stdlib-zulip], home of the stdlib community (see [announcemount][zulip-announcement]). Anyone is welcome to join or browse the web-public channels (which you can see without signing up). The community norms and guidelines provided in this document are to help ensure that everyone can get the most out of this platform and, hopefully, have a positive experience.
```

I don't think it is important to mention that we moved to Zulip in Dec 2025.

</section>

<!-- /.intro -->

## Getting Started

When you first join, a welcome bot will greet you and share some tips specific to stdlib on how to participate effectively. If you're new to Zulip, you may also want to check out [Zulip's official getting started guide][zulip-getting-started] for a quick overview of how the platform works. If you've used other chat platforms, such as Slack or Discord, things should feel pretty familiar. The one key difference is that Zulip organizes conversations into "topics" rather than "threads". This allows for more focused discussions and makes it easier to follow specific conversations without getting lost in a stream of messages. For more on this, see [Zulip's introduction to topics][zulip-intro-to-topics].

We encourage you to come say hello and tell us a bit about yourself in the [**#introductions**][stdlib-introductions] channel.

## Community Norms

### Where to post

> First and foremost, whenever possible, try to post your questions **in public** (rather than sending DMs). Asking questions in public is the fastest way to get a response, as more people can help, _plus_ it's likely that someone else will benefit from finding out the answer to your question. The stdlib team is active in the chat, so you shouldn't need to use **@-mentions** by default.

Reviewer 2: ```suggestion
First and foremost, whenever possible, try to post your questions **in public** (rather than sending direct messages, a.k.a. DMs). Asking questions in public is the fastest way to get a response, as more people can help, _plus_ it's likely that someone else will benefit from finding out the answer to your question. The stdlib team is active in the chat, so you shouldn't need to use **@-mentions** by default.
```

Will everyone know what DM stands for?

Each channel has a description (don't worry too much about posting in the "right" channel—moderators can help [move content][zulip-move-content] if needed), and conversations are organized into [**topics**][zulip-intro-to-topics]. If you're starting a new discussion, try to pick a brief, descriptive topic name that captures the subject of your message. Topic names also help others find relevant information from past conversations (learn more about [Zulip search filters here][zulip-search-filters]). You can edit a topic name at any time, so if you realize that your original topic name isn't quite right, feel free to update it.

Because it's easy to move content around, please **do NOT cross-post**. Not only can this be confusing, but you risk duplication of effort if multiple people respond to the same question in different channels.

### Custom Linkifiers

> We've set up some [custom linkifiers][zulip-linkifiers] to make it easy to share links to GitHub issues, and pull requests. If you type **`#1234`** in a message, it will automatically link to the issue or PR with that number in the main [stdlib GitHub repository][stdlib-github-repo]. There's also a special linkifier for the [stdlib Google Summer of Code repo][stdlib-gsoc-repo], you simply type **`#G{id-number}`**. Note that these work for both issues and pull requests (GitHub uses the same numbering for both).

Reviewer: ```suggestion
We've set up some [custom linkifiers][zulip-linkifiers] to make it easy to share links to GitHub issues and pull requests. If you type **`#1234`** in a message, it will automatically link to the issue or PR with that number in the main [stdlib GitHub repository][stdlib-github-repo]. There's also a special linkifier for the [stdlib Google Summer of Code repo][stdlib-gsoc-repo]; you simply type **`#G{id-number}`**. Note that these work for both issues and pull requests (GitHub uses the same numbering for both).
```

### Code of Conduct

The [stdlib Code of Conduct][stdlib-code-of-conduct] applies to all community spaces, including stdlib's Zulip. Should you encounter an issue, [Zulip's reporting tools][zulip-reporting] and our moderation team will be notified and available to assist.

## Feedback and Suggestions

If you have feedback or suggestions about improving our Zulip community, don't hesitate to share your thoughts. We have a special meta channel, [**#zulip**][stdlib-zulip-channel], for discussing community-related topics. For more general feedback, there's also the aptly-named [**#feedback**][stdlib-feedback-channel] channel.

> The best part of open source is the community, so we're commited to making our Zulip a welcoming and inclusive space for everyone. We look forward to chatting with you there!

Reviewer: Typo: "commited" -> "committed".

```suggestion
The best part of open source is the community, so we're committed to making our Zulip a welcoming and inclusive space for everyone. We look forward to chatting with you there!
```

<section class="notes">

</section>

> <!-- /.notes -->

Reviewer: Same as above — empty `notes` section is a package-README artifact. Drop it when restructuring.

<!--

@license Apache-2.0

Copyright (c) 2026 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

<!-- lint disable expected-html-sections -->

# Zulip Guide

> Guide to the stdlib Zulip chat.

> This guide provides best practices and community norms for the [**stdlib Zulip chat**][stdlib-zulip], home of the stdlib community (see [announcement][zulip-announcement]). Anyone is welcome to join or browse the web-public channels (which you can see without signing up). The community norms and guidelines provided in this document are to help ensure that everyone can get the most out of this platform and, hopefully, have a positive experience.

Reviewer 2: ```suggestion
This guide provides best practices and community norms for the [**stdlib Zulip chat**][stdlib-zulip], home of the stdlib community (see [announcement][zulip-announcement]). Everyone is welcome to join or browse the web-public channels (which you can see without signing up). The community norms and guidelines provided in this document are to help ensure that everyone can get the most out of this platform and, hopefully, have a positive experience.
```

Each channel has a description (don't worry too much about posting in the "right" channel—moderators can help [move content][zulip-move-content] if needed), and conversations are organized into [**topics**][zulip-intro-to-topics]. If you're starting a new discussion, try to pick a brief, descriptive topic name that captures the subject of your message. Topic names also help others find relevant information from past conversations (learn more about [Zulip search filters here][zulip-search-filters]). You can edit a topic name at any time, so if you realize that your original topic name isn't quite right, feel free to update it.

Because it's easy to move content around, please **do NOT cross-post**. Not only can this be confusing, but you risk duplication of effort if multiple people respond to the same question in different channels.

#### Channels

When you first join, you'll automatically be subscribed to several "default" channels. However, there are more channels you might want to explore.

> The channels are roughly organized into categories in the left-hand sidebar. At the bottom, you can click on "Browse Channels" to see the full list.

Reviewer 2: ```suggestion

The channels are roughly organized into categories in the left-hand sidebar. At the bottom, you can click on "Browse Channels" to see the full list.
```

Each channel has a description (don't worry too much about posting in the "right" channel—moderators can help [move content][zulip-move-content] if needed), and conversations are organized into [**topics**][zulip-intro-to-topics]. If you're starting a new discussion, try to pick a brief, descriptive topic name that captures the subject of your message. Topic names also help others find relevant information from past conversations (learn more about [Zulip search filters here][zulip-search-filters]). You can edit a topic name at any time, so if you realize that your original topic name isn't quite right, feel free to update it.

Because it's easy to move content around, please **do NOT cross-post**. Not only can this be confusing, but you risk duplication of effort if multiple people respond to the same question in different channels.

#### Channels

When you first join, you'll automatically be subscribed to several "default" channels. However, there are more channels you might want to explore.

> 
> The channels are roughly organized into categories in the left-hand sidebar. At the bottom, you can click on "Browse Channels" to see the full list.
> 
> Some categories and channels of interest include:
> 
> -   **General Interest**
>     -   **announcements**: Announcements everyone should see, including new releases, blog posts, and public events.
>     -   **general**: General topics.
>     -   **office-hours**: Information related to stdlib's weekly [office hours][stdlib-office-hours].
>     -   **water-cooler**: A place to discuss random topics, unrelated to stdlib and its development.
> 
> -   **Feedback & Support**
>     -   **case-studies**: Using stdlib in your application? Got a cool new demo? Share it here!
>     -   **feedback**: Post your reactions to or feedback on using stdlib.
>     -   **problems**: For reporting and discussing issues in using stdlib.
>     -   **user-questions**: For questions about how to accomplish your goals with stdlib.
> 
> -   **GSOC**
>     -   **gsoc-questions**: Interested in participating in Google Summer of Code? Have questions? Ask here!
> 
> -   **Contributors**
>     -   **dev**: General development topics.
>     -   **dev-questions**: Ask questions related to developing stdlib.
>     -   **getting-started**: Are you a new contributor? Post your questions here!
>     -   **introductions**: Interested in contributing to stdlib? Please introduce yourself here!
>     -   **review-requests**: Requests for PR reviews.
> 
> -   **Development**: Channels related to specific stdlib namespaces, and other areas such as **documentation**, and **tooling**.
> 
> -   **GitHub**: These are feeds generated from GitHub activity, such as new issues and pull requests. You can use these channels to keep up with the latest activity in the project.
> 
> -   **Other**
>     -   **sandbox**: Experiment with Zulip here.
>     -   **zulip**: Questions and discussion about using Zulip.

Reviewer 2: ```suggestion
-   **General Interest**

    -   **announcements**: Announcements everyone should see, including new releases, blog posts, and public events.
    -   **general**: General topics.
    -   **office-hours**: Information related to stdlib's weekly [office hours][stdlib-office-hours].
    -   **water-cooler**: A place to discuss random topics, unrelated to stdlib and its development.

-   **Feedback & Support**

    -   **case-studies**: Using stdlib in your application? Got a cool new demo? Share it here!
    -   **feedback**: Post your reactions to or feedback on using stdlib.
    -   **problems**: For reporting and discussing issues in using stdlib.
    -   **user-questions**: For questions about how to accomplish your goals with stdlib.

-   **GSOC**

    -   **gsoc-questions**: Interested in participating in Google Summer of Code? Have questions? Ask here!

-   **Contributors**

    -   **dev**: General development topics.
    -   **dev-questions**: Ask questions related to developing stdlib.
    -   **getting-started**: Are you a new contributor? Post your questions here!
    -   **introductions**: Interested in contributing to stdlib? Please introduce yourself here!
    -   **review-requests**: Requests for PR reviews.

-   **Development**: Channels related to specific stdlib namespaces, and other areas such as **documentation**, and **tooling**.

-   **GitHub**: These are feeds generated from GitHub activity, such as new issues and pull requests. You can use these channels to keep up with the latest activity in the project.

-   **Other**

    -   **sandbox**: Experiment with Zulip here.
    -   **zulip**: Questions and discussion about using Zulip.
```

    -   **general**: General topics.
    -   **office-hours**: Information related to stdlib's weekly [office hours][stdlib-office-hours].
    -   **water-cooler**: A place to discuss random topics, unrelated to stdlib and its development.

-   **Feedback & Support**

    -   **case-studies**: Using stdlib in your application? Got a cool new demo? Share it here!
    -   **feedback**: Post your reactions to or feedback on using stdlib.
    -   **problems**: For reporting and discussing issues in using stdlib.
    -   **user-questions**: For questions about how to accomplish your goals with stdlib.

-   **GSOC**

    -   **gsoc-questions**: Interested in participating in Google Summer of Code? Have questions? Ask here!

-   **Contributors**

    -   **dev**: General development topics.
    -   **dev-questions**: Ask questions related to developing stdlib.
    -   **getting-started**: Are you a new contributor? Post your questions here!
    -   **introductions**: Interested in contributing to stdlib? Please introduce yourself here!
    -   **review-requests**: Requests for PR reviews.

> -   **Development**: Channels related to specific stdlib namespaces, and other areas such as **documentation**, and **tooling**.

Reviewer 2: ```suggestion
-   **Development**: Channels related to specific stdlib namespaces, and other areas such as documentation and tooling.
```
